### PR TITLE
Declutter logging in RecalculateLaneEndRoutingData

### PR DESCRIPTION
**Pull request by Elesbaan70**
This eliminates 695 lines of code clutter from `RoutingManager.RecalculateLaneEndRoutingData` without actually changing the log output.

### CODE REVIEW

File: TLM/TLM/Manager/Impl/RoutingManager.cs
Comment (kianzarrin) [inline TLM/TLM/Manager/Impl/RoutingManager.cs:1276]: here and in a few other places there is no need for empty line after log.

File: TLM/TLM/Util/DetailLogger.cs
Comment (kianzarrin) [inline TLM/TLM/Util/DetailLogger.cs:10]: we can have a philosophical debate about whether this should be in TLM  or in CSUtils.commons.

Comment (Elesbaan70) [reply TLM/TLM/Util/DetailLogger.cs:10]: I'm following the principle of minimizing the scope until potential outside use cases are identified and better understood.

**Review by kvakvs — APPROVED** [review]
nice

File: TLM/TLM/Util/DetailLogger.cs
Comment (kianzarrin) [inline TLM/TLM/Util/DetailLogger.cs:35]:
```suggestion
             static void BuildBLock(object o) {
```

File: TLM/TLM/Util/DetailLogger.cs
Comment (kianzarrin) [inline TLM/TLM/Util/DetailLogger.cs:40]: The first condition seems redundant.
```suggestion
                    if (p.Name.Any(c => c != '_')) {
```
This lambda is allocating heap per variable but that already happens with input arguments so it is fine.
maybe add a attribute to hide the warning?

File: TLM/TLM/Util/DetailLogger.cs
Comment (kianzarrin) [inline TLM/TLM/Util/DetailLogger.cs:45]:
```suggestion
                    if (v.GetType().IsDefined(typeof(CompilerGeneratedAttribute), false)) {
```

**Review by kianzarrin — CHANGES_REQUESTED** [review]
please fix https://github.com/CitiesSkylinesMods/TMPE/pull/1584#discussion_r886405360 `if (p.Name.Any(c => c != '_')) {`
the rest of the comments are just hints.

File: TLM/TLM/Manager/Impl/RoutingManager.cs
Comment (kianzarrin) [inline TLM/TLM/Manager/Impl/RoutingManager.cs:507]: did you use any automated process to generate output? Maybe instead of checking if every single variable I can check your process.

**Comment by kianzarrin** [comment]
This will create merge conflict with all my PRs. We should wait until my PR's are merged in.

File: TLM/TLM/Manager/Impl/RoutingManager.cs
Comment (Elesbaan70) [reply TLM/TLM/Manager/Impl/RoutingManager.cs:1276]: To me, this enhances readability by visually separating logging code from program logic.

File: TLM/TLM/Util/DetailLogger.cs
Comment (Elesbaan70) [reply TLM/TLM/Util/DetailLogger.cs:35]: I disagree. This just copies the reference into the new stack frame, and doesn't really seem to add any value.

File: TLM/TLM/Util/DetailLogger.cs
Comment (Elesbaan70) [reply TLM/TLM/Util/DetailLogger.cs:50]: This is intended to behave pretty much in a vanilla .net way. While this specific change might seem alright, it would set a precedent of adding a lot of hidden behavior into this class. I would rather leave that up to the caller.

File: TLM/TLM/Manager/Impl/RoutingManager.cs
Comment (Elesbaan70) [reply TLM/TLM/Manager/Impl/RoutingManager.cs:507]: I hardcoded `extendedLogRouting` to `true` and spent a fair bit of time scrutinizing what came out.

File: TLM/TLM/Util/DetailLogger.cs
Comment (Elesbaan70) [reply TLM/TLM/Util/DetailLogger.cs:40]: btw, what heap is being allocated by the lambda?

Comment (kianzarrin) [reply TLM/TLM/Util/DetailLogger.cs:40]: I think making this GC friendly is a waste of time.

File: TLM/TLM/Manager/Impl/RoutingManager.cs
Comment (kianzarrin) [reply TLM/TLM/Manager/Impl/RoutingManager.cs:1276]: i mean the next line is `}`. the problem is that if the log logic take too much space then less of the rest of the logic fit in one page.

File: TLM/TLM/Util/DetailLogger.cs
Comment (Elesbaan70) [reply TLM/TLM/Util/DetailLogger.cs:40]: And now that I'm thinking about it, the tricks won't work anyway because we're using a delegate. 🤷‍♂️

Comment (kvakvs) [reply TLM/TLM/Util/DetailLogger.cs:40]: You guys seem to be having fun making it GC friendly. But its mostly debug-only logging, and its controlled with a switch, so mind that :)

Comment (Elesbaan70) [reply TLM/TLM/Util/DetailLogger.cs:40]: > You guys seem to be having fun making it GC friendly.

I'm really not. 😆 I just did this one optimization because it was convenient and obvious. In general, I'll avoid a heap allocation if there is an obvious way to do so, but that wasn't my main priority here. 😉

**Review by kianzarrin — APPROVED** [review]

**Review by krzychu124 — APPROVED** [review]

**Comment by Elesbaan70** [comment]
@krzychu124 Are any changes still needed related to the build issue you were having?

**Comment by krzychu124** [comment]
> @krzychu124 Are any changes still needed related to the build issue you were having?

I don't think so, but I would like to test Appveyor a little, currently I can't force rebuild to experiment with settings because of the file conflict.

**Pull request merged**
